src/Testcontainers.Pulsar/PulsarBuilder.cs

```diff
@@ -4,7 +4,7 @@ namespace Testcontainers.Pulsar;
 [PublicAPI]
 public sealed class PulsarBuilder : ContainerBuilder<PulsarBuilder, PulsarContainer, PulsarConfiguration>
 {
-    public const string PulsarImage = "apachepulsar/pulsar:3.0.6";
+    public const string PulsarImage = "apachepulsar/pulsar:4.0.0";
```

Comment:
```suggestion
    public const string PulsarImage = "apachepulsar/pulsar:3.0.6";
```

As mentioned in the issue, we don't update the image version unless it is absolutely necessary. We try to maintain backward compatibility.